File: HelloWorld/Procedural.cs
// Compile command: msc Procedural.cs
// Execute commands:
//   mono Procedural
//   mono Procedural "José Carlos Monteiro"
//   mono Procedural Universe

using System;

public class Procedural
{

  public static void Main( string[] arguments )
  {
    string name = ( arguments.Length == 0 ) ? "World" : arguments[ 0 ];
    Console.WriteLine( "Hello, " + name + "!" );

    Console.WriteLine( "» José Carlos Monteiro «" );
  }

}

File: HelloWorld/procedural.go
package main

import (
  "fmt"
  "os"
)

func main() {
  var name string;
  if len( os.Args[ 1: ] ) == 0 {
    name = "World"
  } else {
    name = os.Args[ 1 ]
  }
  fmt.Println( "Hello, " + name + "!" )

  fmt.Println( "» José Carlos Monteiro «" )
}
